Update tool command documentation (#63)
open-eid/libcdoc: doc/tool.md

### Examples

1. **Encrypt a file with a password**  
   Encrypt the file `abc.txt` with the password `Test123`. The resulting container is `abc.txt-pw.cdoc`.

   ```bash
   ./cdoc-tool encrypt --rcpt Test:pw:Test123 --out abc.txt-pw.cdoc ~/Documents/abc.txt
   ```

2. **Encrypt a file with a public key from an ID card**  

>    Encrypt the file `abc.txt` using a public key from an Estonian ID card. The resulting container is `abc.txt-p11pk.cdoc`.

Reviewer: ```suggestion
   Encrypt the file `abc.txt` using a public key from an Estonian ID card. The resulting container is `abc.txt-p11pk.cdoc`. To use the ID card a PKCS11 library has to be specified, the exact location depends on the operating system and installed software.
```
Võib-olla see teeb selgemaks sest paljud ilmselt pole üldse kokku puutunud smart-card'i kasutamisega käsurelat.

   ```bash

>    ./cdoc-tool encrypt --rcpt :p11pk:0:::Isikutuvastus --genlabel --out abc.txt-p11pk.cdoc --library /opt/homebrew/lib/opensc-pkcs11.so ~/Documents/abc.txt

Reviewer: ```suggestion
   ./cdoc-tool encrypt --rcpt :p11pk:0:::Isikutuvastus --genlabel --out abc.txt-p11pk.cdoc --library /opt/homebrew/lib/opensc-pkcs11.so abc.txt
```

   ```

3. **Encrypt a file with a public key from a file**  
   Encrypt the file `abc.txt` using a public key from the file `ec-secp384r1-pub.der`. The resulting container is `abc.txt-pfkey.cdoc`.

   ```bash

>    ./cdoc-tool encrypt --rcpt :pfkey:ec-secp384r1-pub.der --genlabel --out abc.txt-pfkey.cdoc ~/Documents/abc.txt

Reviewer: ```suggestion
   ./cdoc-tool encrypt --rcpt :pfkey:ec-secp384r1-pub.der --genlabel --out abc.txt-pfkey.cdoc abc.txt
```

   ```

4. **Encrypt a file with an AES key**  

>    Encrypt the file `abc.txt` using an AES key provided via the command line. The resulting container is `abc.txt-aes.cdoc`.

Reviewer: ```suggestion
   Encrypt the file `abc.txt` using an AES key provided on the command line. The resulting container is `abc.txt-aes.cdoc`.
```
Jälle minu keeletunne, milles ma pole absoluutselt kindel

   ```bash

>    ./cdoc-tool encrypt --rcpt Test:skey:E165475C6D8B9DD0B696EE2A37D7176DFDF4D7B510406648E70BAE8E80493E5E --genlabel --out abc.txt-aes.cdoc ~/Documents/abc.txt

Reviewer: ```suggestion
   ./cdoc-tool encrypt --rcpt Test:skey:E165475C6D8B9DD0B696EE2A37D7176DFDF4D7B510406648E70BAE8E80493E5E --genlabel --out abc.txt-aes.cdoc abc.txt
```

   ```

5. **Encrypt a file with a public key from an ID card and use key server**  
   Encrypt the file `abc.txt` using a public key from an Estonian ID card and use the RIA key server. The resulting container is `abc.txt-ks.cdoc`.

   ```bash

>    ./cdoc-tool encrypt --rcpt Test:p11pk:0:::Isikutuvastus --library /opt/homebrew/lib/opensc-pkcs11.so --server 00000000-0000-0000-0000-000000000000 https://cdoc2.id.ee:8443 --accept keyserver-cert.der --out abc.txt-ks.cdoc ~/Documents/abc.txt

Reviewer: ```suggestion
   ./cdoc-tool encrypt --rcpt Test:p11pk:0:::Isikutuvastus --library /opt/homebrew/lib/opensc-pkcs11.so --server 00000000-0000-0000-0000-000000000000 https://cdoc2.id.ee:8443 --accept keyserver-cert.der --out abc.txt-ks.cdoc abc.txt
```

   ```

---

## Decryption

> The syntax for decrypting an encrypted file is as follows:

Reviewer: The following?

### Examples

1. **Decrypt a file with a password**  

>    Decrypt the file `abc.txt-pw.cdoc` using the key label `Test` and password `Test123`.

Reviewer: ```suggestion
   Decrypt the file `abc.txt-pw.cdoc` using the key with label `Test` and password `Test123`.
```
Tehniliselt korrektne oleks "the key in the lock with label..." aga see on liiga kohmakas

   ```bash
   ./cdoc-tool decrypt --label Test --password Test123 abc.txt-pw.cdoc
   ```

2. **Decrypt a file with an ID card**  

>    Decrypt the file `abc.txt-p11pk.cdoc` using the key label index `1` and an Estonian ID card with PIN code `1234`.

Reviewer: ```suggestion
   Decrypt the file `abc.txt-p11pk.cdoc` using the key from lock `1` and an Estonian ID card with PIN code `1234`.
```
See "ke label index" on väga ebaõnnestunud sõnastus ja tgelt on ka argumendi nimi imalik

   ```bash
   ./cdoc-tool decrypt --label_idx 1 --pin 1234 --slot 0 --key-label Isikutuvastus --library /opt/homebrew/lib/opensc-pkcs11.so abc.txt-p11pk.cdoc
   ```

3. **Decrypt a file with an ID card and use key server**  

>    Decrypt the file `abc.txt-ks.cdoc` using the key label `Test` and a private key from an ID card and using the RIA key server.

Reviewer: ```suggestion
   Decrypt the file `abc.txt-ks.cdoc` using the key with label `Test` and a private key from an ID card, using the RIA key server.
```
Õkskõik kuidas ma proovin seda ümber sõnastada, ikka on jube kohmakas
